### DISCUSSION TITLE
How to set $EDITOR for Windows git bash properly

### DISCUSSION THREAD

**Discussion by Skrelpoid** [jj-vcs/jj#2027] [Q&A]
When I run jj describe, I get "Failed to run editor pico".

When I set EDITOR to nano in my windows environment variables, running jj describe does not open nano but instead just kind of freezes/let's me type into the next line, but I need to CTRL+C to get out and it does not save the message.

I tried setting EDITOR to /usr/bin/nano but this causes  Failed to run editor 'C:/Program Files/Git/usr/bin/nano'

How can I properly set the EDITOR for windows git bash? If nano doesn't work I am also fine with notepad++"

**Comment by martinvonz** [comment]
Where is the `nano` executable stored? I'm guessing `C:/Program Files/Git/usr/bin/nano` isn't the correct path.

Also, `pico` (and `nano`) are probably terrible defaults on Windows. As a Windows user, can you tell us what a good default would be? It's ideally something that's installed by default. My guess is that that leaves us with Notepad as the only choice.

**Reply by Skrelpoid** [reply]
Actually nano is installed there, it seems Windows doesn't play with it for some reason. Somewhere on stack overflow I saw a workaround but I haven't tried it yet. I will try to find it and let you know if it worked.

And I guess yeah notepad as a default on windows would be fine.

**Reply by yuja** [reply]
Wild guess: `bin/nano` isn't an `.exe` but a `.bat` (or some kind of wrapper), and Windows can't execute it without using `cmd /c`?

Maybe related: https://github.com/martinvonz/jj/issues/1966

**Comment by Skrelpoid** [comment]
nano indeed was an exe. But I found this stack overflow answer: https://stackoverflow.com/questions/10564/how-can-i-set-up-an-editor-to-work-with-git-on-windows/10625#10625
that said on windows you should use start /wait in the EDITOR variable.

I created  a script and put it in C:\startwaitNano.cmd with following content:
`start /wait /b "nano" "C:\Program Files\Git\usr\bin\nano.exe" %*`
(/b is for staying in the same window, the first quoted string "nano" is the window title for some reason, which does not apply with /b)

I set EDITOR to C:\startWaitNano.cmd 
and now jj describe works

I also tried notepad.exe instead of nano.exe and it worked as well

**Reply by yuja** [reply]
Good to know that, thanks. Windows CLI is magic to me.

Maybe you can also inline the script to `.jjconfig.toml` if you like.
```toml
[ui]
editor = ['start', '/wait', '/b', 'nano', 'C:\Program Files\Git\usr\bin\nano.exe']
```
